blocks/library/tag-cloud/term-list.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * Calculate font size for terms based on their post count.
+ *
+ * @param {Array} terms The list of terms.
+ *
+ * @return {Array} Array of terms with font scale.
+ */
+function calculateFontSizes( terms ) {
+	const MIN = 8;
+	const MAX = 22;
+	const UNIT = 'pt';
+
+	// Sort terms by post count ascending
+	const sortedTerms = terms.slice();
+	sortedTerms.sort( ( a, b ) => a.count - b.count );
+
+	// Get first and last items in array to know
+	// the lowest and highest post count in list
+	const lowestCount = sortedTerms[ 0 ].count;
+	const highestCount = sortedTerms.pop().count;
+
+	const range = Math.max( highestCount - lowestCount, 1 );
+	const fontStep = ( MAX - MIN ) / range;
+
+	// Calculate font size for each term
+	terms.forEach( ( term, i ) => {
+		const countWeight = term.count - lowestCount;
+		const fontSize = MIN + ( countWeight * fontStep );
+		terms[ i ].fontScale = `${ fontSize }${ UNIT }`;
```

Comment:
You're mutating the original `terms` reference array here which, based on your usage of the return value `termFontSizes` below, I don't think is expected.

This might be a good candidate for an [`Array#map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) (returning an array of font sizes) or [`Array#reduce`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce) (returning an object of term name => font size).

```js
// Calculate font size for each term
return terms.map( ( term, i ) => {
	const countWeight = term.count - lowestCount;
	const fontSize = MIN + ( countWeight * fontStep );
	return fontSize;
} );
```

Reply:
Thanks for pointing it out, I chose to use `map` since the terms order stays consistent I figured it would make the code simpler.